proxy-node-gateway/src/main/java/uk/gov/ida/notification/resources/EntityMetadata.java

```diff
@@ -0,0 +1,30 @@
+package uk.gov.ida.notification.resources;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class EntityMetadata {
+
+    enum Key {encryptionCertificate, eidasDestination}
+
+    private final Map<String, Map<Key, String>> data;
+
+    public EntityMetadata() {
+        this.data = new HashMap<>();
+    }
+
+    public synchronized void setData(String issuer, Key key, String value) {
+        Map<Key, String> entry = data.getOrDefault(issuer, new HashMap());
+        entry.put(key, value);
+        data.put(issuer, entry);
+    }
+
+    public synchronized String getValue(String issuer, Key key) {
```

Comment:
I think this might leave us with a few issues that having a redis solved.
- If we have more than one instance of the gateway running, if the gateway receives a hub response and it's processed by an instance that hasn't yet seen an authn request from that issuer, this will throw an IllegalStateException. Assuming high traffic from issuers this should be fairly rare, but it's possible.
- If we release the gateway at any point this map will be lost, and hub responses will throw an IllegalStateException (if no new authn requests have been seen from that issuer by the gateway instance processing the request)

Could we store the encryption cert and the destination in the cookie with the other attributes? I'm not sure what the storage limit is...